src/main/java/io/airlift/slice/SizeOf.java

```diff
@@ -86,6 +90,11 @@ public static long sizeOf(Object[] array)
         return (array == null) ? 0 : sizeOfObjectArray(array.length);
     }
 
+    public static long sizeOf(String string)
+    {
+        return (string == null) ? 0 : (STRING_INSTANCE_SIZE + string.length() * Character.BYTES);
```

Comment:
> `string.length() * Character.BYTES`

Starting with Java 9, this is an invalid assumption to make.

https://openjdk.java.net/jeps/254

Reply:
This seems reasonable as an upper bound. Is there a way to access the compact size?

Reply:
Unfortunately, I don’t think so. If we want to make it an upper bound, we should rename it. All other variants produce a guaranteed size, so it’d be misleading to name it the same way (e.g., estimatedSizeOf)

Reply:
We have this logic in a bunch of places in Presto, so we need this as a functionality, in a single place.
Of course, if this isn't precise, we can name it differently.

Is `estimatedSizeOf` something we agree upon?